crates/cli/src/main.rs

```diff
@@ -528,6 +537,97 @@ async fn start_program(opt: Cli) -> Result<(), AnvilZksyncError> {
             }
         }
     }
+
+    // Use `debug-trace` command to get debug info for a transaction
+    // TODO: clean this up and move to a separate function?
+    if let Command::DebugTrace(args) = command {
+        let params = if args.only_top {
+            json!([args.tx, { "tracer": "callTracer", "tracerConfig": { "onlyTopCall": true } }])
+        } else {
+            json!([args.tx])
+        };
+
+        let rpc_url = args.rpc_url.to_config().url.to_string();
+        let result_value: serde_json::Value =
+            match rpc_call::<serde_json::Value>(&rpc_url, "debug_traceTransaction", params).await {
+                Ok(v) => v,
+                Err(e) => {
+                    sh_println!("debug_traceTransaction failed: {e}");
+                    return Ok(());
+                }
+            };
+
+        if result_value.is_null() {
+            sh_println!("No debug info found for tx {:#x}", args.tx);
+            return Ok(());
+        }
+
+        // TODO: add improved halt/revert handling
+        let exec_result = {
+            let out_bytes: Vec<u8> = result_value
+                .get("output")
+                .and_then(Value::as_str)
+                .and_then(|s| s.parse::<Bytes>().ok())
+                .map(|b| b.to_vec())
+                .unwrap_or_default();
+
+            if let Some(err) = result_value.get("error").and_then(Value::as_str) {
+                ExecutionResult::Halt {
+                    reason: Halt::TracerCustom(err.to_string()),
+                }
+            } else if result_value
+                .get("reverted")
+                .and_then(Value::as_bool)
+                .unwrap_or(false)
+            {
+                ExecutionResult::Revert {
+                    output: VmRevertReason::General {
+                        msg: format_revert_reason_hex(&out_bytes),
+                        data: out_bytes.clone(),
+                    },
+                }
+            } else {
+                ExecutionResult::Success { output: out_bytes }
+            }
+        };
+        let tx_result_for_arena = VmExecutionResultAndLogs::mock(exec_result);
+
+        if args.raw {
+            sh_println!(
+                "Raw debug payload:\n{}",
+                serde_json::to_string_pretty(&result_value).unwrap_or_default()
+            );
```

Comment:
Nit: consider adding padding, feel free to ignore if there is more important stuff to worry about :)